woocommerce/woocommerce-ios
# [Product Multi-Selection] Remove selections on clear selection

Closes: #8973

## DESCRIPTION

This PR adds back "Clear Selection" back to both the Product selector and the ProductVariation selector, which were removed temporarily for Product Multi-Selection iteration 1.

## Changes:
- Switches the existing Configuration's `isClearSelectionEnabled` to true when the Product Multi-Selection feature is enabled (as Single-Selection does not use it).
- Adds a `onClearedSelection` callback to the `ProductSelectorViewModel`'s initializer, which is called when we tap on "Clear Selection".

This is necessary to clear the `selectedProducts` and `selectedProductVariations` published properties from the `EditableOrderViewModel`, as these are passed as initial selected items into the `ProductSelectorViewModel` when is initialized and then the different views are re-rendered, causing a bug where variations are never properly unselected because they're persisted and passed through the different objects.

## Testing instructions
- Go to Menu > Settings > Experimental features > and enable Product Multi-Selection
- Go to Orders > Tap `+` > Tap on `+ Add Products` > Tap on different products and variations and tap on `Clear Selection`.
- See how products are unselected.
